src/Features/Core/Portable/UseThrowExpression/UseThrowExpressionCodeFixProvider.cs

```diff
@@ -54,13 +55,25 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                 var throwStatementExpression = root.FindNode(diagnostic.AdditionalLocations[1].SourceSpan);
                 var assignmentValue = root.FindNode(diagnostic.AdditionalLocations[2].SourceSpan);
 
-                // First, remote the if-statement entirely.
+                var text = document.GetTextSynchronously(cancellationToken);
+                var throwLine = text.Lines.GetLineFromPosition(diagnostic.AdditionalLocations[1].SourceSpan.Start);
+
+                var fullThrowStatement = root.FindNode(throwLine.SpanIncludingLineBreak, getInnermostNodeForTie: true);
+
+                var triviaList = new SyntaxTriviaList()
+                    .AddRange(fullThrowStatement.GetLeadingTrivia())
+                    .AddRange(fullThrowStatement.GetTrailingTrivia());
+
+                // First, remove the if-statement entirely.
                 editor.RemoveNode(ifStatement);
 
-                // Now, update the assignment value to go from 'a' to 'a ?? throw ...'.
-                editor.ReplaceNode(assignmentValue,
-                    generator.CoalesceExpression(assignmentValue,
-                    generator.ThrowExpression(throwStatementExpression)));
+                var newNode = generator.CoalesceExpression(assignmentValue,
+                    generator.ThrowExpression(throwStatementExpression))
+                    .WithTrailingTrivia(triviaList).NormalizeWhitespace();
```

Comment:
one problem is that you're adding the trailing trivia to the throw expression.  that's not correct (and is why the trivia is after the `)` but before the `;`).  You want to add the trailing trivia *after* the semicolon/statement that is being generated.